Tests/Microsoft.AppCenter.Test.Functional/AppCenter/AppCenterCoreTest.cs

```diff
@@ -12,33 +12,38 @@ namespace Microsoft.AppCenter.Test.Functional.AppCenter
     using Analytics = Microsoft.AppCenter.Analytics.Analytics;
     using AppCenter = Microsoft.AppCenter.AppCenter;
 
-    public class AppCenterCoreTest : IDisposable
+    public class AppCenterCoreTest
     {
         // Before
         public AppCenterCoreTest()
         {
-            Utils.deleteDatabase();
-        }
-
-        // After
-        public void Dispose()
-        {
-            // Let pending SDK calls be completed.
-            Task.Delay(3000).Wait();
+            Utils.DeleteDatabase();
         }
 
         [Fact]
         public async Task EnableDisableTest()
         {
+            var httpNetworkAdapter = new HttpNetworkAdapter();
+            DependencyConfiguration.HttpNetworkAdapter = httpNetworkAdapter;
+            var startServiceTask = httpNetworkAdapter.MockRequestByLogType("startService");
+            var startServiceTask2 = httpNetworkAdapter.MockRequestByLogType("startService");
+            var startServiceTask3 = httpNetworkAdapter.MockRequestByLogType("startService");
+
             // Start App Center.
             AppCenter.UnsetInstance();
             Analytics.UnsetInstance();
             AppCenter.LogLevel = LogLevel.Verbose;
-            AppCenter.Start(Config.resolveAppsecret(), typeof(Analytics));
+            AppCenter.Start(Config.ResolveAppSecret(), typeof(Analytics));
+
+            // Wait for "startService" log to be sent.
+            Task.WaitAny(startServiceTask, Task.Delay(5000));
 
             // Disable Appcenter.
             await AppCenter.SetEnabledAsync(false);
 
+            // On iOS when set disabled all appcenter logs are removing from DB. We should wait here otherwise there might be a deadlock.
+            Task.Delay(3000).Wait();
```

Comment:
I guess that the reason can be the same as in

https://github.com/microsoft/appcenter-sdk-apple/blob/3c4e38fe929e9982eb845f5463253aacd73bca8a/Sasquatch/SasquatchUITests/AppCenterUITests.swift#L89-L94

Reply:
From the logs, I see that if executed without delay the code is deadlocked (trying to remove all logs from the DB and then access the removed logs), so I'm not sure that it has something to do with NSUserDefaults .